File: 942_DI_String_Match.java
class Solution {
    public int[] diStringMatch(String S) {
        int low = 0, high = S.length();
        List<Integer> res = new ArrayList<>();
        for (int i = 0; i < S.length(); i++) {
            if (S.charAt(i) == 'I') {
                res.add(low++);
            } else {
                res.add(high--);
            }
        }
        res.add(low);
        return res.stream().mapToInt(i -> i).toArray();
    }
}

File: python/942_DI_string_match.py
class Solution:
    def diStringMatch(self, S: str) -> List[int]:
        num_I = 0
        num_D = len(S)
        res = []
        for c in S:
            if c == 'I':
                res.append(num_I)
                num_I += 1
            elif c == 'D':
                res.append(num_D)
                num_D -= 1
        res.append(num_I)
        return res

